Dockerfile

```diff
@@ -1,68 +1,145 @@
+
+# -------------------------------------------------------------------------------------------------
+#
+# final layer
+# use the smallest debain with headless openjdk and copying files from download layers
+FROM openjdk:8-jre-slim
+
+ARG VCS_REF
+ARG GRAYLOG_VERSION
+ARG BUILD_DATE
+ARG GRAYLOG_HOME=/usr/share/graylog
+ARG GRAYLOG_USER=graylog
+ARG GRAYLOG_UID=1100
+ARG GRAYLOG_GROUP=graylog
+ARG GRAYLOG_GID=1100
+
+# hadolint ignore=DL3023
+COPY --from=graylog-downloader /opt/graylog ${GRAYLOG_HOME}
+COPY config ${GRAYLOG_HOME}/data/config
+
+WORKDIR ${GRAYLOG_HOME}
+
+# hadolint ignore=DL3008,DL3014,SC2086
+RUN \
+  echo "export JAVA_HOME=/usr/lib/jvm/java-8-openjdk-amd64/jre" > /etc/profile.d/graylog.sh && \
+  echo "export BUILD_DATE=${BUILD_DATE}"           >> /etc/profile.d/graylog.sh && \
+  echo "export GRAYLOG_VERSION=${GRAYLOG_VERSION}" >> /etc/profile.d/graylog.sh && \
+  echo "export GRAYLOG_SERVER_JAVA_OPTS='-XX:+UnlockExperimentalVMOptions -XX:+UseCGroupMemoryLimitForHeap -XX:NewRatio=1 -XX:MaxMetaspaceSize=256m -server -XX:+ResizeTLAB -XX:+UseConcMarkSweepGC -XX:+CMSConcurrentMTEnabled -XX:+CMSClassUnloadingEnabled -XX:+UseParNewGC -XX:-OmitStackTraceInFastThrow'" >> /etc/profile.d/graylog.sh && \
+  echo "export GRAYLOG_HOME=${GRAYLOG_HOME}"       >> /etc/profile.d/graylog.sh && \
+  echo "export GRAYLOG_USER=${GRAYLOG_USER}"       >> /etc/profile.d/graylog.sh && \
+  echo "export GRAYLOG_GROUP=${GRAYLOG_GROUP}"     >> /etc/profile.d/graylog.sh && \
+  echo "export GRAYLOG_UID=${GRAYLOG_UID}"         >> /etc/profile.d/graylog.sh && \
+  echo "export GRAYLOG_GID=${GRAYLOG_GID}"         >> /etc/profile.d/graylog.sh && \
+  echo "export PATH=${GRAYLOG_HOME}/bin:${PATH}"   >> /etc/profile.d/graylog.sh && \
+  apt-get update  > /dev/null && \
+  apt-get install --no-install-recommends --assume-yes \
+    curl \
+    'gosu=1.10-*' \
+    libcap2-bin > /dev/null && \
+  addgroup \
+    --gid "${GRAYLOG_GID}" \
+    --quiet \
+    "${GRAYLOG_GROUP}" && \
+  adduser \
+    --disabled-password \
+    --disabled-login \
+    --gecos '' \
+    --home ${GRAYLOG_HOME} \
+    --uid "${GRAYLOG_UID}" \
+    --gid "${GRAYLOG_GID}" \
+    --quiet \
+    "${GRAYLOG_USER}" && \
+  chown --recursive "${GRAYLOG_USER}":"${GRAYLOG_GROUP}" ${GRAYLOG_HOME} && \
+  setcap 'cap_net_bind_service=+ep' "${JAVA_HOME}/bin/java" && \
+  apt-get remove --assume-yes --purge \
+    apt-utils > /dev/null && \
+  rm -f /etc/apt/sources.list.d/* && \
+  apt-get clean > /dev/null && \
+  apt autoremove --assume-yes > /dev/null && \
+  rm -rf \
+    /tmp/* \
+    /var/cache/debconf/* \
+    /var/lib/apt/lists/* \
+    /var/log/* \
+    /usr/share/X11 \
+    /usr/share/doc/* 2> /dev/null
+
+
+COPY docker-entrypoint.sh /
+
+EXPOSE 9000
+VOLUME ${GRAYLOG_HOME}/data
+ENTRYPOINT ["/docker-entrypoint.sh"]
+CMD ["graylog"]
+
+# add healthcheck
+HEALTHCHECK \
+  --interval=10s \
+  --timeout=2s \
+  --retries=12 \
+  CMD curl --silent --fail http://localhost:9000/api/ || exit 1
```

Comment:
what does happen if someone use other port than 9000 for the webinterface?

Reply:
Well, then we have to make it configurable.
Otherwise the HEALTHCHECK won't work.
But, the configuration can be integrated via a `--volume`. This would not make this HEALTHCHACK work. 
But I would have a solution for that.

Reply:
that would be awesome - it is very common to have Graylog not on port `9000`